### PR TITLE
Renaming Sync metrics to prevent name collision

### PR DESCRIPTION
**Pull request by skhamis**
To solve the issues detailed in: https://bugzilla.mozilla.org/show_bug.cgi?id=1827949

In short:
Glean will run into name collisions with moving the metrics to application-services. So we're renaming the metrics in`metrics.yaml`  to `*_v2` to prevent this. Since we now control the wrappers there is surprisingly-little churn for the API since most of the renaming affects under-the-hood` but the top-level APIs might be relatively unchanged.

Part 2 of: https://github.com/mozilla/application-services/pull/5588 but didn't want to update it since it was already approved (also might be easier to split the changes

### Pull Request checklist ###

- **Breaking changes**:  This PR follows our [breaking change policy](https://github.com/mozilla/application-services/blob/main/docs/howtos/breaking-changes.md)
  - [ ] This PR follows the breaking change policy:
     - This PR has no breaking API changes, or
     - There are corresponding PRs for our consumer applications that resolve the breaking changes and have been approved
- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGELOG.md](../CHANGELOG.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

**Review by bendk — DISMISSED** [review]
This is on top of https://github.com/mozilla/application-services/pull/5588 right? 

The renaming looks good to me

**Review by lougeniaC64 — DISMISSED** [review]
LGTM

**Comment by skhamis** [comment]
### Request for data collection review form

All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.
<ol dir="auto">
<li>What questions will you answer with this data?</li>
</ol>
<p dir="auto">
This is a rename of the existing telemetry (on fenix and iOS) to consolidate our telemetry into one place and have a much easier time managing sync telemetry./p>
<ol dir="auto" start="2">
<li>Why does Mozilla need to answer these questions?  Are there benefits
 for users? Do we need this information to address product or business 
requirements? Some example responses:</li>
</ol>
<p dir="auto">Sync is one of the core features of our products; we need 
to know that the implementations of it that we're shipping are healthy.</p>
<ol dir="auto" start="3">
<li>What alternative methods did you consider to answer these questions? Why were they not sufficient?</li>
</ol>
<p dir="auto">The alternatives is to have iOS and Android having separate telemetry. However this potentially leaves into branching of telemetry which produces holes in our understanding of sync health and where issues might be occurring.</p>
<ol dir="auto" start="4">
<li>
<p dir="auto">Can current instrumentation answer these questions?<br>
Sort of; we can see how individual data type syncs are behaving, but we 
can't quite tell if Sync overall is functioning correctly, since the 
current telemetry doesn't allow us to tie together pings for different 
data type syncs which all run together.</p>
</li>
<li>
<p dir="auto">List all proposed measurements and indicate the category of data collection for each measurement, using the <a href="https://wiki.mozilla.org/Firefox/Data_Collection" rel="nofollow">Firefox data collection categories</a> found on the Mozilla wiki.</p>
</li>
</ol>

**Since this is a rename of the existing telemetry. I will pull the data collection categories that it was shipped with.**

Measurement Description | Data Collection Category | Tracking Bug #
-- | -- | --
Global sync ping, which includes a global failureReason and a global syncUuid unique to a single Sync | Category 1 | https://github.com/mozilla-mobile/android-components/issues/5371
history-sync.syncUuid metric, same value as the global syncUuid | Category 1 |
bookmarks-sync.syncUuid metric, same value as the global syncUuid | Category 1 |
logins-sync.syncUuid metric, same value as the global syncUuid | Category 1 |
addresses-sync.syncUuid metric, same value as the global syncUuid | Category 1 |
creditcards-sync.syncUuid metric, same value as the global syncUuid | Category 1 |
tabs-sync.syncUuid metric, same value as the global syncUuid | Category 1 |


<ol dir="auto" start="6">
<li>How long will this data be collected?  Choose one of the following:</li>
</ol>
<p dir="auto">I want to permanently monitor this data. Sammy Khamis + <code class="notranslate">application-services</code> team.</p>
<ol dir="auto" start="7">
<li>What populations will you measure?</li>
</ol>
<p dir="auto">All populations for all sync users using iOS and Android clients</p>
<ol dir="auto" start="8">
<li>If this data collection is default on, what is the opt-out mechanism for users?</li>
</ol>
<p dir="auto">Standard Glean SDK mechanism.</p>
<ol dir="auto" start="9">
<li>Please provide a general description of how you will analyze this data.</li>
</ol>
<p dir="auto">Redash/GLAM (when available) dashboards.</p>
<ol dir="auto" start="10">
<li>Where do you intend to share the results of your analysis?</li>
</ol>
<p dir="auto">Within the Sync, Android and iOS teams.</p>
<ol dir="auto" start="11">
<li>Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection? If so:</li>
</ol>

**Comment by eliserichards** [comment]
> ### Request for data collection review form
> 
> All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.
> 
>     1. What questions will you answer with this data?
> 
> 
> This is a rename of the existing telemetry (on fenix and iOS) to consolidate our telemetry into one place and have a much easier time managing sync telemetry./p>
> 
>     2. Why does Mozilla need to answer these questions?  Are there benefits
>         for users? Do we need this information to address product or business 
>        requirements? Some example responses:
> 
> 
> Sync is one of the core features of our products; we need to know that the implementations of it that we're shipping are healthy.
> 
>     3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
> 
> 
> The alternatives is to have iOS and Android having separate telemetry. However this potentially leaves into branching of telemetry which produces holes in our understanding of sync health and where issues might be occurring.
> 
>     4. Can current instrumentation answer these questions?
>        Sort of; we can see how individual data type syncs are behaving, but we 
>        can't quite tell if Sync overall is functioning correctly, since the 
>        current telemetry doesn't allow us to tie together pings for different 
>        data type syncs which all run together.
> 
>     5. List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Firefox/Data_Collection) found on the Mozilla wiki.
> 
> 
> **Since this is a rename of the existing telemetry. I will pull the data collection categories that it was shipped with.**
> Measurement Description 	Data Collection Category 	Tracking Bug #
> Global sync ping, which includes a global failureReason and a global syncUuid unique to a single Sync 	Category 1 	[mozilla-mobile/android-components#5371](https://github.com/mozilla-mobile/android-components/issues/5371) and https://bugzilla.mozilla.org/show_bug.cgi?id=1827949
> history-sync-v2.syncUuid metric, same value as the global syncUuid 	Category 1 	[mozilla-mobile/android-components#5371](https://github.com/mozilla-mobile/android-components/issues/5371) and https://bugzilla.mozilla.org/show_bug.cgi?id=1827949
> bookmarks-sync-v2.syncUuid metric, same value as the global syncUuid 	Category 1 	[mozilla-mobile/android-components#5371](https://github.com/mozilla-mobile/android-components/issues/5371) and https://bugzilla.mozilla.org/show_bug.cgi?id=1827949
> logins-sync-v2.syncUuid metric, same value as the global syncUuid 	Category 1 	[mozilla-mobile/android-components#5371](https://github.com/mozilla-mobile/android-components/issues/5371) and https://bugzilla.mozilla.org/show_bug.cgi?id=1827949
> addresses-sync-v2.syncUuid metric, same value as the global syncUuid 	Category 1 	[mozilla-mobile/android-components#5371](https://github.com/mozilla-mobile/android-components/issues/5371) and https://bugzilla.mozilla.org/show_bug.cgi?id=1827949
> creditcards-sync-v2.syncUuid metric, same value as the global syncUuid 	Category 1 	[mozilla-mobile/android-components#5371](https://github.com/mozilla-mobile/android-components/issues/5371) and https://bugzilla.mozilla.org/show_bug.cgi?id=1827949
> tabs-sync-v2.syncUuid metric, same value as the global syncUuid 	Category 1 	[mozilla-mobile/android-components#5371](https://github.com/mozilla-mobile/android-components/issues/5371) and https://bugzilla.mozilla.org/show_bug.cgi?id=1827949
> fxa_tab_v2.sent metric, recorded when a tab is sent 	Category 1 	[mozilla-mobile/android-components#5371](https://github.com/mozilla-mobile/android-components/issues/5371)
> fxa_tab_v2.received metric, recorded when a tab is sent 	Category 1 	[mozilla-mobile/android-components#5371](https://github.com/mozilla-mobile/android-components/issues/5371)
> 
>     6. How long will this data be collected?  Choose one of the following:
> 
> 
> I want to permanently monitor this data. Sammy Khamis + `application-services` team.
> 
>     7. What populations will you measure?
> 
> 
> All populations for all sync users using iOS and Android clients
> 
>     8. If this data collection is default on, what is the opt-out mechanism for users?
> 
> 
> Standard Glean SDK mechanism.
> 
>     9. Please provide a general description of how you will analyze this data.
> 
> 
> Redash/GLAM (when available) dashboards.
> 
>     10. Where do you intend to share the results of your analysis?
> 
> 
> Within the Sync, Android and iOS teams.
> 
>     11. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection? If so:
> 
> 
> No.

# Data Review Form (to be filled by Data Stewards)

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 

* Yes, through `metrics.yaml`, `pings.yaml`, and the [Glean Dictionary](https://dictionary.telemetry.mozilla.org/)

2) Is there a control mechanism that allows the user to turn the data collection on and off? (Note, for data collection not needed for security purposes, Mozilla provides such a control mechanism) Provide details as to the control mechanism available.

* Yes, through the Standard Glean SDK mechanism.

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

* Yes, this data will be permanently collected. Sammy Khamis (skhamis@mozilla.com) will be monitoring along with the sync team (sync-team@mozilla.com).

4) Using the **[category system of data types](https://wiki.mozilla.org/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

* Category 1 - technical data.

5) Is the data collection request for default-on or default-off?

* Default on.

6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?

* No. 

7) Is the data collection covered by the existing Firefox privacy notice? **If unsure: escalate to legal if:**

* Yes.

8) Does the data collection use a third-party collection tool? **If yes, escalate to legal.**

* No.

## Result

data-review+

**Comment by skhamis** [comment]
sorry I know it got approved but I accidentally switched the base branch (was trying to do that to another PR) which I think reset the approvals -- I'll re-request review @bendk or @lougeniaC64 is fine

File: components/sync_manager/metrics.yaml [hunk 1]
```diff
@@ -28,6 +28,7 @@ sync_v2:
       - technical
     notification_emails:
       - sync-team@mozilla.com
+      - skhamis@mozilla.com
```
Comment (lougeniaC64) [inline components/sync_manager/metrics.yaml:31]: nit: Aren't you already included in [sync-team@mozilla.com(https://groups.google.com/a/mozilla.com/g/sync-team/members)?

Comment (eliserichards) [reply components/sync_manager/metrics.yaml:31]: Yup! If it's permanently monitored we want a single owner who can take responsibility 😬

**Pull request merged**